contracts/core/Registry.sol

```diff
@@ -21,27 +38,77 @@ contract Registry {
     // attestationAddr = addr(hash(id, name, owner))
     // OR deploy right away with CREATE2 as proxy so they can do anything (or just forward ERC20s and ETH)
     // id => IdentityDetails
-    function createIdentity(IdentityDetails memory _identityDetails, address[] memory _owners) external returns (uint256) {
 
+    // ASK: changing the function signature as attestationAddr has to be generated from the name and caller
+    /// @notice Creates a new identity
+    /// @dev This will also set the attestation address generated from msg.sender and name
+    function createIdentity(
+        string memory name,
+        Metadata memory metadata,
+        address[] memory _owners
+    ) external returns (uint256) {
+
+        IdentityDetails memory identityDetails = IdentityDetails(
+            name,
+            metadata,
+            // ASK : should identityId be used as the hash for attestationAddr?
+            // cause project can't have same attestAddr across chains
+            generateAttestationAddr(name, msg.sender)
+        );
+
+        identities[identityId++] = identityDetails;
+
+        for(uint i = 0; i < _owners.length; i++) {
+            _grantRoles(_owners[i], identityId);
+        }
+        
+        emit IdentityCreated(identityId, name, metadata, identityDetails.attestationAddr);
+        
+        return identityId;
+    }
+
+    /// @notice Updates the name of the identity
+    /// @dev This will also change the attestation address, since it's a hash that includes the name
+    // ASK: should only coreOwner be able to change name?
```

Comment:
^ @zobront @codenamejason

Reply:
I think there may be more than 1 person we want to give the ability to change the name.

I'd use the OWNERS in the roles the ability to do this, but not the MEMBERS.